File: src/Application Specific/Classes/Entity Collections/Spares/PartSupplier/PartSupplierValidator.vb
Namespace Entity

Namespace PartSuppliers

Public Class PartSupplierValidator
    'make sure that contact object is valid
    Inherits BaseValidator

    Public Sub Validate(ByVal oPartSupplier As PartSupplier)

                'make sure that contact object is valid
                If oPartSupplier.Errors.Count > 0 Then
                    Dim de As DictionaryEntry
                    For Each de In oPartSupplier.Errors
                        Me.AddCriticalMessage(CStr(de.Value))
                    Next
                End If

                'If oPartSupplier.PartCode.Trim.Length = 0 Then
                '    Me.AddCriticalMessage("Part Code Missing")
                'End If

                If oPartSupplier.SupplierID <= 0 Then
                    Me.AddCriticalMessage("Supplier Missing")
                End If

                If oPartSupplier.QuantityInPack < 1 Then
                    Me.AddCriticalMessage("Invalid Quantity In Pack")
                End If

                If oPartSupplier.Price <= 0 Then
                    Me.AddCriticalMessage("Invalid Price")
                End If

                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

            End Sub

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Spares/PartSupplier/PartSupplierValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace PartSuppliers
    {
        // make sure that contact object is valid
        public class PartSupplierValidator : BaseValidator
        {
            public void Validate(PartSupplier oPartSupplier)
            {
                // make sure that contact object is valid
                if (oPartSupplier.Errors.Count > 0)
                {
                    foreach (DictionaryEntry de in oPartSupplier.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                // If oPartSupplier.PartCode.Trim.Length = 0 Then
                // Me.AddCriticalMessage("Part Code Missing")
                // End If

                if (oPartSupplier.SupplierID <= 0)
                {
                    AddCriticalMessage("Supplier Missing");
                }

                if (oPartSupplier.QuantityInPack < 1)
                {
                    AddCriticalMessage("Invalid Quantity In Pack");
                }

                if (oPartSupplier.Price <= 0)
                {
                    AddCriticalMessage("Invalid Price");
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}